### PR TITLE
Implemented "wasp build" cmd - generates different schema.prisma.

### PR DESCRIPTION
**Pull request by matijaSos**

### CODE REVIEW

File: waspc/cli/Command/Build.hs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+build = do
+    waspProjectDir <- findWaspProjectRootDirFromCwd
+    let outDir = waspProjectDir </> Common.dotWaspDirInWaspProjectDir
+                 </> Common.buildDirInDotWaspDir
+
+    waspSaysC "Building wasp project..."
+    buildResult <- liftIO $ buildIO waspProjectDir outDir
+    case buildResult of
+        Left compileError -> throwError $ CommandError $ "Build failed: " ++ compileError
+        Right () -> waspSaysC "Code has been successfully built!\n"
```
Comment (Martinsos) [inline waspc/cli/Command/Build.hs:26]: Maybe mention location where it can be found now.

File: waspc/cli/Command/Common.hs [hunk 2]
```diff
@@ -1,24 +1,49 @@
...
                                          dotWaspRootFileInWaspProjectDir,
                                          waspSays)
+import qualified Lib
+import           CompileOptions         (CompileOptions (..))
 import           StrongPath             (Abs, Dir, Path)
 import qualified StrongPath             as SP
+import qualified Util.IO
 
+compileIOWithOptions :: CompileOptions
+                    -> Path Abs (Dir WaspProjectDir)
```
Comment (Martinsos) [inline waspc/cli/Command/Common.hs:30]: Indent for one more space.

File: waspc/cli/Command/Common.hs [hunk 3]
```diff
@@ -1,24 +1,49 @@
...
+                    -> Path Abs (Dir Lib.ProjectRootDir)
+                    -> IO (Either String ())
+compileIOWithOptions options waspProjectDir outDir = do
```
Comment (Martinsos) [inline waspc/cli/Command/Common.hs:33]: This one should probably go under Command/Compile ?

File: waspc/src/Generator/DbGenerator.hs [hunk 4]
```diff
@@ -37,17 +38,26 @@ dbSchemaFileInDbRootDir = SP.castRel dbSchemaFileInDbTemplatesDir
...
-genPrismaSchema entities = createTemplateFileDraft dstPath tmplSrcPath (Just templateData)
+genPrismaSchema :: Wasp -> FileDraft
+genPrismaSchema wasp = createTemplateFileDraft dstPath tmplSrcPath (Just templateData)
     where
         dstPath = dbSchemaFileInProjectRootDir
         tmplSrcPath = dbTemplatesDirInTemplatesDir </> dbSchemaFileInDbTemplatesDir
 
+        isBuild = Wasp.getIsBuild wasp
+
+        datasourceProvider = if isBuild then "postgresql" else "sqlite"
```
Comment (Martinsos) [inline waspc/src/Generator/DbGenerator.hs:56]: Cooler might be if you do:
```hs
(datasourceProvider, datasourceUrl) = if isBuild then ("postgresql", "env(\"DATABASE_URL\")")
                                                 else ("sqlite",     "\"file:./dev.db\"")
```

File: waspc/src/Lib.hs [hunk 5]
```diff
@@ -24,8 +24,14 @@ compile waspFile outDir options = do
...
     case Parser.parseWasp waspStr of
         Left err -> return $ Left (show err)
-        Right wasp -> do
-            externalCodeFiles <- ExternalCode.readFiles (CompileOptions.externalCodeDirPath options)
-            generateCode $ wasp `setExternalCodeFiles` externalCodeFiles
+        Right wasp -> expandWaspASTWithCompileOptions wasp options >>= generateCode
   where
     generateCode wasp = Generator.writeWebAppCode wasp outDir options >> return (Right ())
+    
+expandWaspASTWithCompileOptions :: Wasp -> CompileOptions -> IO Wasp
```
Comment (Martinsos) [inline waspc/src/Lib.hs:31]: Ok this part is slightly weird.
You say you expand it with CompileOptions, but you also add external files -> so the name is not completely correct.
Also, I find it unusual that you are replicating stuff from compile options in Wasp -> why do we need to have isBuild in both places? Why not just add CompileOptions as one of the properties of Wasp, for now?

**Review by Martinsos — APPROVED** [review]
Approved, I really like how you did this! But some comments are probably worth taking a look! Also  tests are not passing at the moment.

File: waspc/src/Lib.hs (hunk 5 above)
Comment (matijaSos) [reply waspc/src/Lib.hs:31]: resolved, changed the name to better reflect that not only copying is happening.

File: waspc/src/Generator/DbGenerator.hs (hunk 4 above)
Comment (matijaSos) [reply waspc/src/Generator/DbGenerator.hs:56]: really cool! :D

**Pull request merged**